src/Neuron/Zettelkasten/Markdown/Extension.hs

```diff
@@ -0,0 +1,48 @@
+{-# LANGUAGE OverloadedStrings #-}
+
+module Neuron.Zettelkasten.Markdown.Extension
+  ( customRender,
+  )
+where
+
+import Control.Monad (forM_)
+import Data.List.NonEmpty (NonEmpty (..))
+import qualified Data.List.NonEmpty as NE
+import Lucid
+import Text.MMark.Extension (Block (Table), Extension)
+import qualified Text.MMark.Extension as Ext
+
+newline :: Html ()
+newline = "\n"
+
+-- https://github.com/mmark-md/mmark/blob/8f5534d8068c2b7a139b893639ee5920bcaedd84/Text/MMark/Render.hs#L111-L136
+customRender :: Extension
+customRender = Ext.blockRender $ \old block ->
+  case block of
+    (Table calign (hs :| rows)) -> do
+      table_ [class_ "ui celled table"] $ do
+        newline
+        thead_ $ do
+          newline
+          tr_
+            $ forM_ (NE.zip calign hs)
+            $ \(a, h) ->
+              th_ (alignStyle a) (snd h)
+          newline
+        newline
+        tbody_ $ do
+          newline
+          forM_ rows $ \row -> do
+            tr_
+              $ forM_ (NE.zip calign row)
+              $ \(a, h) ->
+                td_ (alignStyle a) (snd h)
+            newline
+        newline
+      newline
+    other -> old other
+  where
+    alignStyle Ext.CellAlignDefault = []
+    alignStyle Ext.CellAlignLeft = [style_ "text-align:left"]
+    alignStyle Ext.CellAlignRight = [style_ "text-align:right"]
+    alignStyle Ext.CellAlignCenter = [style_ "text-align:center"]
```

Comment:
Is our only choice here to fork [upstream code](https://github.com/mmark-md/mmark/blob/8f5534d8068c2b7a139b893639ee5920bcaedd84/Text/MMark/Render.hs#L111-L136) just to modify the class name? Is there a way to re-use upstream code without having to fork it?

Reply:
This is a good point. I opened the PR in this state to get the conversation started but I'm not happy to fork the code either.

I took a (quick) look at mmark and didn't find a way to do it. But extending the way certain blocks are rendered seems common enough and could be covered by mmark I guess.

Reply:
> extending the way certain blocks are rendered seems common enough and could be covered by mmark I guess

To be precise, it is supported by `Ext.blockRender`. But it would be great to not have to copy the whole rendering of the table just to add a class. I guess that's what you also meant.

Reply:
Mark's [suggestion](https://github.com/mmark-md/mmark/issues/70#issuecomment-613639892) to use [`with`](https://hackage.haskell.org/package/lucid-2.9.12/docs/Lucid.html#v:with) seems like it would obviate the need to fork the renderer.

Reply:
I'll try that. 👌

Reply:
Much better now!